text/0000-sawtooth-sdk-swift.md

```diff
@@ -0,0 +1,214 @@
+Sawtooth client:
+- Create a web app using the Javascript SDK
+- Implement custom methods to create and sign transactions natively
+
+The first solution does not work for a use case that requires a native
+application. The second solution raises the barrier of entry for developers
+with a background in mobile development who want to use Sawtooth. The creation
+of the Sawtooth Swift SDK, including documentation and examples, will make the
+Sawtooth platform more accessible to those coming from mobile development.
+
+# Prior art
+[prior-art]: #prior-art
+
+The implementation of the SawtoothSigning module for the Swift SDK is based
+on existing implementations for other languages, such as 
+[Java](https://github.com/hyperledger/sawtooth-sdk-java/tree/master/sawtooth-sdk-signing/src/main/java/sawtooth/sdk/signing) 
+and
+[Rust](https://github.com/hyperledger/sawtooth-sdk-rust/blob/master/src/signing/secp256k1.rs).
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+### To be implemented
+
+- To use Carthage to import the SawtoothSigning framework into an iOS project
+  it is necessary to include a Cartfile, which describes the project’s
+  dependencies to Carthage. GitHub repositories are specified with the github
+  keyword. For example, it is possible to load the framework from a branch:
+  ```
+  github "bitwiseio/sawtooth-sdk-swift" "master"
+  ```
+
+  Or from a release:
+  ```
+  github "bitwiseio/sawtooth-sdk-swift" ~> 0.1
+  ```
+
+  If a new repository within the Hyperledger organization is created to host
+  the Sawtooth Swift SDK, the existing Carfile in the XO example needs to be
+  updated for the new path of the SDK. The documentation also needs to be
+  updated to include instructions on how to import the SawtoothSigning
+  framework from the new hyperledger repository.
+
+- The Sawtooth protos are not included as part of the Swift SDK. Before
+  stabilization, we may want to include them as part of the SDK. Currently,
+  the Sawtooth `.proto` files and Swift protogen scripts are included directly
+  in the XO client example application. Further work must be done to
+  implement the generation of the protobuf messages as part of the Swift SDK.
+
+### Other issues
+
+- The documentation for the Swift SDK uses the same templates as the other
+  SDKs. The templates were copied from the sawtooth-core repository. This is
+  not an ideal solution, as there is an extra overhead to maintaining several
+  versions of the same documentation across several repositories.
+
+- Ideally, the documentation for the Swift SDK should be published under the
+  Application Developer's Guide section in the Sawtooth documentation,
+  together with the documentation for the other SDKs. It is not clear what the
+  best strategy is to publish the Swift SDK documentation, given that the
+  source files and code base are in an different repository than the Sawtooth
+  documentation.
```

Comment:
The other SDKs appear with the Sawtooth core documentation. This has the advantage of being a lot easier to find and access if with the other docs.
The reason the other SDKs were split out was for sawtooth-core build issues and extra time required.

Reply:
The docs should be contained in the swift sdk repository as they need to be maintained in-sync with the source code. We are working on a mechanism to combine docs during publishing, since this isn't the only instance this issue exists.

Reply:
I agree with both @danintel and @vaporos: The core Sawtooth doc should make it easy to find all SDK docs, and a future mechanism for "combined docs" is highly desirable. But that's outside the scope of this RFC.  

For the Swift SDK, the following Sawtooth core topics could easily be updated to include info about and a link to the Swift SDK docs. 
- Available SDKs: https://sawtooth.hyperledger.org/docs/core/nightly/master/app_developers_guide/sdk_table.html
- Using the Sawtooth SDKs: 
https://sawtooth.hyperledger.org/docs/core/nightly/master/app_developers_guide/using_the_sdks.html

Reply:
I shortened the description of this issue to the core problem. It is largely out of scope for the RFC, but still important to mention as it will affect the proposed repository later on.